homeassistant/components/wsdot/config_flow.py

```diff
@@ -0,0 +1,166 @@
+            step_id=SOURCE_USER,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_API_KEY): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_import(self, import_info: dict[str, Any]) -> ConfigFlowResult:
+        """Handle a flow initialized by import."""
+        await self.async_set_unique_id(import_info[CONF_API_KEY])
+        self._abort_if_unique_id_configured()
+        wsdot_travel_times = wsdot_api.WsdotTravelTimes(import_info[CONF_API_KEY])
+        try:
+            travel_time_routes = await wsdot_travel_times.get_all_travel_times()
+        except wsdot_api.WsdotTravelError as ws_error:
+            if ws_error.status == 400:
+                return self.async_abort(reason="invalid_api_key")
+            return self.async_abort(reason="cannot_connect")
+
+        subentries = []
+        for route in import_info[CONF_TRAVEL_TIMES]:
+            maybe_travel_time = [
+                tt
+                for tt in travel_time_routes
+                # old platform configs could store the id as either a str or an int
+                if str(tt.TravelTimeID) == str(route[CONF_ID])
+            ]
+            if not maybe_travel_time:
+                _LOGGER.error(
+                    "Found legacy WSDOT travel_time that does not describe a valid travel_time route (%s)",
+                    route,
+                )
+                continue
+            travel_time = maybe_travel_time[0]
+            route_name = travel_time.Name
+            unique_id = "_".join(travel_time.Name.split())
+            subentries.append(
+                ConfigSubentry(
+                    subentry_type=SUBENTRY_TRAVEL_TIMES,
+                    unique_id=unique_id,
+                    title=route_name,
+                    data=MappingProxyType(
+                        {CONF_NAME: travel_time.Name, CONF_ID: travel_time.TravelTimeID}
+                    ),
+                ).as_dict()
+            )
+
+        return self.async_create_entry(
+            title=DOMAIN,
+            data={
+                CONF_API_KEY: import_info[CONF_API_KEY],
+            },
+            subentries=subentries,
+        )
+
+    @classmethod
+    @callback
+    def async_get_supported_subentry_types(
+        cls, config_entry: ConfigEntry
+    ) -> dict[str, type[ConfigSubentryFlow]]:
+        """Return subentries supported by wsdot."""
+        return {SUBENTRY_TRAVEL_TIMES: TravelTimeSubentryFlowHandler}
+
+
+class TravelTimeSubentryFlowHandler(ConfigSubentryFlow):
+    """Handle subentry flow for adding WSDOT Travel Times."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> SubentryFlowResult:
+        """Add a new Travel Time subentry."""
+        runtime = self._get_entry().runtime_data
+        if runtime is None:
+            raise ConfigEntryError("WSDOT entry has no runtime_data")
```

Comment:
Either
1. Check if the entry is loaded and return self.async_abort
2. Just create a new client, which detaches your config flow from the state of the integration itself